src/fe-common/core/completion.c

```diff
@@ -241,28 +241,37 @@ char *word_complete(WINDOW_REC *window, const char *line, int *pos, int erase, i
 	if (complist == NULL)
 		return NULL;
 
+	/* escape if the word doesn't begin with '/' and expand_escapes are turned on */
+	data = *line != '/' && settings_get_bool("expand_escapes") ?
+		escape_string(complist->data) : complist->data;
+
 	/* word completed */
-	*pos = startpos+strlen(complist->data);
+	*pos = startpos+strlen(data);
 
 	/* replace the word in line - we need to return
 	   a full new line */
 	result = g_string_new(line);
 	g_string_erase(result, startpos, wordlen);
-	g_string_insert(result, startpos, complist->data);
+	g_string_insert(result, startpos, data);
 
 	if (want_space) {
 		if (!isseparator(result->str[*pos]))
 			g_string_insert_c(result, *pos, ' ');
 		(*pos)++;
 	}
 
-	wordlen = strlen(complist->data);
+	wordlen = strlen(data);
 	last_line_pos = *pos;
 	g_free_not_null(last_line);
 	last_line = g_strdup(result->str);
 
 	ret = result->str;
 	g_string_free(result, FALSE);
+
+	/* free data if we have been escaping something */
+	if (data != complist->data)
```

Comment:
I'm not a great fan of such pointer comparisons, some other places in the code handle this situation by adding a `g_strdup` and always call `g_free` when they're done.